internal/pkg/db/redis/data.go

```diff
@@ -14,15 +14,29 @@
 package redis
 
 import (
+	"fmt"
+	"time"
+
+	"github.com/edgexfoundry/go-mod-core-contracts/clients/logger"
+	contract "github.com/edgexfoundry/go-mod-core-contracts/models"
+
 	correlation "github.com/edgexfoundry/edgex-go/internal/pkg/correlation/models"
 	"github.com/edgexfoundry/edgex-go/internal/pkg/db"
-	contract "github.com/edgexfoundry/go-mod-core-contracts/models"
 
 	"github.com/gomodule/redigo/redis"
 	"github.com/google/uuid"
 	"github.com/imdario/mergo"
 )
 
+// deleteReadingsChannel channel used to delete readings asynchronously
+var deleteReadingsChannel = make(chan string, 50)
+var deleteEventsChannel = make(chan string, 50)
+
+const (
+	// Batch size used when pipelining asynchronous deletions
+	AsyncDeleteBatchSize = 1000
+)
```

Comment:
Get rid of all of these global variables. See comment above w/r/t how the channels can be managed as members of the Redis client. `AsyncDeleteBatchSize` could even be a configuration setting that is fed to Redis.NewClient()